src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -951,6 +951,22 @@ public function withCount($relations)
         return $this->withAggregate(is_array($relations) ? $relations : func_get_args(), '*', 'count');
     }
 
+        /**
+     * Add a relationship count conditionally, with an optional alias.
+     *
+     * @param  string  $relation
+     * @param  Closure  $closure
```

Comment:
You could narrow this more down if you want:
```suggestion
     * @param  Closure(\Illuminate\Database\Eloquent\Builder): mixed  $closure
```